tracing/span_test.go

```diff
@@ -292,7 +292,7 @@ func TestSpanTypes(t *testing.T) {
 			func(t *testing.T) {
 				t.Parallel()
 				span := Span{Name: "test", spanType: c.spanType}
-				if span.Type() != c.expected {
+				if span.Type().String() != c.expected {
```

Comment:
Here we should make `c.expected` be of type `SpanType` instead of string, and compare `span.Type() != c.expected` instead (and in `Errorf` below use `%v` instead of `%s`).

Reply:
In this case, what we really want to test is the switch statement in `SpanType.String`, so that's not quite the test we want.  I think we can just take the `Span` out of the test entirely though and do this:

```golang
func(t *testing.T) {
	t.Parallel()
	if c.spanType.String() != c.expected {
		t.Errorf(
			"Expected SpanType.String to be %s, got %v",
			c.expected,
			c.spanType,
		)
	}
}
```